src/apis/btc-core.ts

```diff
@@ -125,14 +130,32 @@ export class DefaultBTCCoreAPI implements BTCCoreAPI {
             console.log(`Error during tx lookup by OP_RETURN: ${e}`);
         }
 
-        if (!txs.length) {
-            return Promise.reject("No transaction id found");
-        }
-        if (txs.length > 1) {
-            return Promise.reject("OP_RETURN collision detected");
+        for (const tx of txs) {
+            if (tx.vout === undefined) {
+                continue;
+            }
+            for (const vout of tx.vout) {
+                if (this.txOutputHasRecipientAndAmount(vout, recipientAddress, amountAsBTC)) {
+                    return tx.txid;
+                }
+            }
         }
+        return Promise.reject("No transaction id found");
+    }
 
-        return txs[0].txid;
+    private txOutputHasRecipientAndAmount(vout: VOut, recipientAddress?: string, amountAsBTC?: string): boolean {
+        if (recipientAddress) {
+            if (recipientAddress !== vout.scriptpubkey_address) {
+                return false;
+            }
+            if (amountAsBTC) {
+                const amountAsSatoshi = btcToSat(amountAsBTC);
+                if (amountAsSatoshi !== vout.value?.toString()) {
```

Comment:
We allow users on the parachain to "overpay" the vault. So instead of checking for equality here, please check if the sent amount is greater than the expected amount.